pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -643,6 +643,32 @@ func adsClientCount() int {
 	return n
 }
 
+func (s *DiscoveryServer) ProxyUpdate(clusterID, ip string) {
+	var connection *XdsConnection
+
+	adsClientsMutex.RLock()
+	// Create a temp map to avoid locking the add/remove
+	for _, v := range adsClients {
+		if v.modelNode.ClusterID == clusterID && v.modelNode.IPAddresses[0] == ip {
+			connection = v
+			break
+		}
+
+	}
+	adsClientsMutex.RUnlock()
+
+	currentlyPending := s.pushQueue.Pending()
+	if currentlyPending != 0 {
+		adsLog.Infof("Starting new push while %v were still pending", currentlyPending)
```

Comment:
nit - I think this is misleading, this happens during the middle of pushed at any time so it's not uncommon for this to occur

Reply:
I copied from the previous one, i will use debug here